tests/vxlan/test_vxlan_vnet_bgp_subintf.py

```diff
@@ -0,0 +1,450 @@
+        "ports": ports,
+        "stage": "egress",
+        "type": ACL_TYPE_NAME
+    }
+    acl_update["ACL_RULE"][f"{ACL_TABLE_NAME}|rule_1"] = {
+        "INNER_SRC_IP": wl_bgp_ip,
+        "INNER_SRC_MAC_REWRITE_ACTION": INNER_SRC_MAC,
+        "TUNNEL_VNI": VNI,
+        "PRIORITY": "1005"
+    }
+
+    logger.info("Programming acl table type onto DUT: " + str(acl_update))
+    duthost.copy(content=json.dumps(acl_update, indent=2), dest="/tmp/acl_update.json")
+    duthost.shell("sonic-cfggen -j /tmp/acl_update.json --write-to-db")
+
+    time.sleep(5)  # wait for acl to be applied
+
+
+def setup_vnet_routes(duthost, wl_intf_info, vnet_name):
+    config_update = {
+        "VNET_ROUTE_TUNNEL": {
+            f"{vnet_name}|150.0.0.0/24": {
+                "endpoint": TUNNEL_ENDPOINT
+            }
+        },
+        "VNET_ROUTE": {
+            f"{vnet_name}|{SUBNET_RANGE}": {
+                "nexthop": wl_intf_info["wl_bgp_ip"],
+                "ifname": wl_intf_info["wl_portchannel"]
+            }
+        }
+    }
+
+    logger.info("Programming vnet routes onto DUT: " + str(config_update))
+    duthost.copy(content=json.dumps(config_update, indent=2), dest="/tmp/vnet_route_update.json")
+    duthost.shell("sonic-cfggen -j /tmp/vnet_route_update.json --write-to-db")
+
+
+def setup_bgp_and_interfaces(duthost, vnet_name, bgp_neighs):
+    wl_portchannel = None
+    wl_portchannel_ip = None
+    wl_bgp_name = None
+    wl_bgp_ip = None
+
+    ip_int = duthost.show_and_parse("show ip int")
+    for entry in ip_int:
+        bgp_neigh = entry.get("bgp neighbor", "N/A")
+        if bgp_neigh != "" and bgp_neigh != "N/A":
+            wl_bgp_name = bgp_neigh
+            wl_portchannel = entry.get("interface")
+            wl_portchannel_ip = entry.get("ipv4 address/mask")
+            wl_bgp_ip = entry.get("neighbor ip")
+            break
+
+    pytest_assert(wl_portchannel is not None or wl_bgp_name is not None,
+                  "Cannot find a portchannel with a bgp neighbor.")
+    pytest_assert(wl_portchannel_ip is not None,
+                  f"Cannot find IP address for portchannel {wl_portchannel}.")
+    pytest_assert(wl_bgp_ip is not None,
+                  f"Cannot find BGP neighbor IP for BGP session {wl_bgp_name}.")
+
+    # Remove existing BGP neighbor
+    asn = bgp_neighs[wl_bgp_ip]["asn"]
+    duthost.shell(f"sonic-db-cli CONFIG_DB DEL 'BGP_NEIGHBOR|{wl_bgp_ip}'")
+
+    # Add BGP peer range and put portchannel in vnet
+    bgp_peers = {
+        f"{vnet_name}|WLPARTNER_PASSIVE_V4": {
+            "ip_range": [wl_portchannel_ip],
+            "name": "WLPARTNER_PASSIVE_V4",
+            "peer_asn": asn,
+            "src_address": wl_portchannel_ip.split('/')[0]
+        }
+    }
+    portchannel_intfs = {
+        wl_portchannel: {
+            "vnet_name": vnet_name
+        }
+    }
+
+    config_update = {
+        "BGP_PEER_RANGE": bgp_peers,
+        "PORTCHANNEL_INTERFACE": portchannel_intfs,

```

Comment:
I am surprised that the addition of vnet later works, I recall in the past that if a portchannel was created without vnet added and then the vnet was added later it resulted in a failure stating that vnet association cannot be modified, can you double check that this test works consistently and does not run into this condition from time to time.